lib/azure/armrest/insights/event_service.rb

```diff
@@ -44,22 +44,16 @@ def initialize(armrest_configuration, options = {})
         #   filter = "eventTimestamp ge #{date} and eventChannels eq 'Admin, Operation'"
         #   select = "resourceGroupName, operationName"
         #
-        #   ies.list(:filter => filter, :select => select, :all => true).events.each{ |event|
+        #   ies.list(:filter => filter, :select => select, :all => true).each{ |event|
         #     p event
         #   }
         #
         def list(options = {})
           url = build_url(options)
           response = rest_get(url)
-          json_response = JSON.parse(response.body)
 
-          events = ArmrestCollection.new(
-            json_response['value'].map do |hash|
-              Azure::Armrest::Insights::Event.new(hash)
-            end
-          )
-
-          events.continuation_token = parse_skip_token(json_response)
+          klass  = Azure::Armrest::Insights::Event
```

Comment:
My thought is for `ArmrestCollection#initialize()` 
```
  def initialize(array, header = nil)
    # parse and process header
    super(array)
  end
```
then you only needs to append like this
```
events = ArmrestCollection.new(
  json_response['value'].map { |hash| Azure::Armrest::Insights::Event.new(hash) },
  response.header
)
```